### PR TITLE
Add .NET 6 support for iOS/macOS

### PR DESCRIPTION
**Pull request by MatkovIvan**
* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [ ] Are the files formatted correctly?
* [ ] Did you add unit tests if this modifies the Windows code?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Add .NET 6 support

## Related PRs or issues

[AB#93424](https://dev.azure.com/msmobilecenter/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/93424)
[AB#93611](https://dev.azure.com/msmobilecenter/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/93611)
[AB#93612](https://dev.azure.com/msmobilecenter/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/93612)

### CODE REVIEW

File: SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.iOS/Microsoft.AppCenter.Crashes.iOS.csproj [hunk 1]
```diff
@@ -21,13 +21,12 @@
     <DefineConstants>DEBUG;</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <MtouchLink>
-    </MtouchLink>
-    <MtouchHttpClientHandler>
-    </MtouchHttpClientHandler>
+    <MtouchLink>SdkOnly</MtouchLink>
```
Comment (DmitriyKirakosyan) [inline SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.iOS/Microsoft.AppCenter.Crashes.iOS.csproj:25]: Why it should be SdkOnly?

File: build.cake [hunk 2]
```diff
@@ -101,25 +106,31 @@ Task("Externals-Apple")
     .WithCriteria(() => IsRunningOnUnix())
     .Does(() =>
 {
-    CleanDirectory(AppleExternals);
     var zipFile = System.IO.Path.Combine(AppleExternals, "apple.zip");
+    if (FileExists(zipFile))
```
Comment (DmitriyKirakosyan) [inline build.cake:110]: This will skip the download part if zip exists. Is there a task which re-downloads the archive?

File: test-tools.cake [hunk 3]
```diff
@@ -1,10 +1,9 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-#tool nuget:?package=XamarinComponent
-#addin nuget:?package=Cake.Xamarin
-#addin nuget:?package=Cake.FileHelpers&version=3.0.0
-#addin nuget:?package=Newtonsoft.Json
+#addin nuget:?package=Cake.Xamarin&version=4.0.0
+#addin nuget:?package=Cake.FileHelpers&version=5.0.0
+#addin nuget:?package=Newtonsoft.Json&version=13.0.1
```
Comment (DmitriyKirakosyan) [inline test-tools.cake:6]: Why should we specify Newtonsoft.Json version explicitly now?

File: SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.iOS/Microsoft.AppCenter.Crashes.iOS.csproj (hunk 1 above)
Comment (MatkovIvan) [reply SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.iOS/Microsoft.AppCenter.Crashes.iOS.csproj:25]: VS wants to declare it explicitly

File: SDK/AppCenterAnalytics/Microsoft.AppCenter.Analytics.MacOS/Microsoft.AppCenter.Analytics.MacOS.csproj [hunk 4]
```diff
@@ -30,9 +30,10 @@
     <IncludeMonoRuntime>false</IncludeMonoRuntime>
     <UseSGen>false</UseSGen>
     <HttpClientHandler />
-    <LinkMode />
+    <LinkMode>None</LinkMode>
```
[the remark this answers is not in the document]
Comment (MatkovIvan) [reply SDK/AppCenterAnalytics/Microsoft.AppCenter.Analytics.MacOS/Microsoft.AppCenter.Analytics.MacOS.csproj:34]: VS wants to declare it explicitly

File: build.cake (hunk 2 above)
Comment (MatkovIvan) [reply build.cake:110]: not really. It long task and if you want to really re-download it, you should clean first

File: test-tools.cake (hunk 3 above)
Comment (MatkovIvan) [reply test-tools.cake:6]: New version of Cake forces it

File: SDK/AppCenterAnalytics/Microsoft.AppCenter.Analytics.AppleUnified/StructsAndEnums.cs [hunk 5]
```diff
@@ -1,8 +1,6 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-using System;
-
-namespace Microsoft.AppCenter.Analytics.iOS.Bindings
+namespace Microsoft.AppCenter.Analytics.Apple.Bindings
```
[the remark this answers is not in the document]
Comment (MatkovIvan) [reply SDK/AppCenterAnalytics/Microsoft.AppCenter.Analytics.AppleUnified/StructsAndEnums.cs:4]: Yes, it's default bindings file

File: SDK/AppCenterAnalytics/Microsoft.AppCenter.Analytics.AppleUnified/Analytics.cs [hunk 6]
```diff
@@ -54,15 +53,15 @@ public static Task SetEnabledAsync(bool enabled)
...
         /// <summary>
         /// Resume the Analytics service.
         /// </summary>
         internal static void Resume()
         {
-            iOSAnalytics.Resume();
+            MSACAnalytics.Resume();
         }
 
         /// <summary>
```
[the remark this answers is not in the document]
Comment (MatkovIvan) [reply SDK/AppCenterAnalytics/Microsoft.AppCenter.Analytics.AppleUnified/Analytics.cs:67]: It's here from `0.*`, don't touch the legacy 😅

**Review by DmitriyKirakosyan — APPROVED** [review]

**Pull request merged**